shell/src/main/java/alluxio/shell/command/TestCommand.java

```diff
@@ -0,0 +1,124 @@
+ */
+
+package alluxio.shell.command;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.URIStatus;
+import alluxio.exception.AlluxioException;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Tests properties of the path specified in args.
+ */
+@ThreadSafe
+public final class TestCommand extends AbstractShellCommand {
+
+  /**
+   * @param fs the filesystem of Alluxio
+   */
+  public TestCommand(FileSystem fs) {
+    super(fs);
+  }
+
+  @Override
+  public String getCommandName() {
+    return "test";
+  }
+
+  @Override
+  public int getNumOfArgs() {
+    return 1;
+  }
+
+  @Override
+  protected Options getOptions() {
+    return new Options()
+        .addOption(DIR_OPTION)
+        .addOption(FILE_OPTION)
+        .addOption(PATH_EXIST_OPTION)
+        .addOption(DIR_NOT_EMPTY_OPTION)
+        .addOption(FILE_ZERO_LENGTH_OPTION);
+  }
+
+  private void printResult(boolean testResult) {
+    if (testResult) {
+      System.out.println(0);
+    } else {
+      System.out.println(1);
+    }
+  }
+
+  /**
+   * Tests properties of the path specified in args.
+   *
+   * @param path The {@link AlluxioURI} path as the input of the command
+   * @param testDir test whether the path is a directory
+   * @param testFile test whether the path is a file
+   * @param testPathExist test whether the path exists
+   * @param testDirNotEmpty test whether the directory is not empty
+   * @param testFileZeroLength test whether the file is zero length
+   * @throws AlluxioException when Alluxio exception occurs
+   * @throws IOException when non-Alluxio exception occurs
+   */
+  public void test(AlluxioURI path, boolean testDir, boolean testFile, boolean testPathExist,
+                   boolean testDirNotEmpty, boolean testFileZeroLength)
+                   throws AlluxioException, IOException {
+    try {
+      URIStatus status = mFileSystem.getStatus(path);
+      boolean testResult = false;
+      if (testDir && status.isFolder()) {
+        testResult = true;
+      } else if (testFile && !status.isFolder()) {
+        testResult = true;
+      } else if (testPathExist) {
+        testResult = true;
+      } else if (testDirNotEmpty && status.isFolder() && status.getLength() > 0) {
+        testResult = true;
+      } else if (testFileZeroLength && !status.isFolder() && status.getLength() == 0) {
+        testResult = true;
+      }
+      printResult(testResult);
+    } catch (AlluxioException | IOException e) {
+      System.out.println(1);
+    }
+  }
+
+  @Override
+  public void run(CommandLine cl) throws AlluxioException, IOException {
+    String[] args = cl.getArgs();
+    AlluxioURI path = new AlluxioURI(args[0]);
+    test(path, cl.hasOption("d"), cl.hasOption("f"), cl.hasOption("e"),
```

Comment:
If the user specify more than one option, then in AlluxioShell.java, when executes "CommandLine cmdline = command.parseAndValidateArgs(args)", it will directly throw an exception and will not run this testcommand. So I think there is no need to make sure the user didn't specify more than one option here.